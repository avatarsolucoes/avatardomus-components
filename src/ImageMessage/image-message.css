
.imageMessage,
.imageMessage .imageContent,
.imageMessage .messageText{
  display: block;
  max-width: 100%;
  position: relative;
  text-align: center;
  margin: 0 auto;
}

.imageMessage .messageText {
  width: 100%;
}

.imageMessage svg,
.imageMessage img{
  max-width: 100%;
  width: 200px;
}


