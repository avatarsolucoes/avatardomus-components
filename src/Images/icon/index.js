import componentList from './component';
import requireList from './require';
import importList from './import';

/**
 * @function getImageByType
 * @param {String} typeImage
 * @param {Boolean} isSvg
 * @returns {import('react').ReactSVG|String}
 */
export function getImageByType(typeImage, isSvg) {
  if (!isSvg) {
    return typeImage in componentList ? componentList[typeImage] : componentList.Unknow;
  }
  return typeImage in requireList ? requireList[typeImage] : requireList.unknow;
}

const Icon = {
  getImageByType,
  componentList,
  requireList,
  importList,
};

export default Icon;
