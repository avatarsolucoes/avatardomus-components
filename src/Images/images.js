import ImagesSvgRequire from './imagesSvgRequire';
import ImagesSvgImport from './imagesSvgImport';
import ImagesAsComponent from './imagesComponent';

export const svgImportList = {
  ...ImagesSvgImport,
};

export const svgRequireList = {
  ...ImagesSvgRequire,
};

export const svgComponentList = {
  ask: ImagesAsComponent.Ask,
  blocked: ImagesAsComponent.Blocked,
  building: ImagesAsComponent.Building,
  click: ImagesAsComponent.Click,
  forbidden: ImagesAsComponent.Forbidden,
  locked: ImagesAsComponent.Locked,
  nofound: ImagesAsComponent.Nofound,
  offline: ImagesAsComponent.Offline,
  sad: ImagesAsComponent.Sad,
  scripting: ImagesAsComponent.Scripting,
  timeout: ImagesAsComponent.Timeout,
  vehicle: ImagesAsComponent.Vehicle,
  unknow: ImagesAsComponent.Unknow,
};
