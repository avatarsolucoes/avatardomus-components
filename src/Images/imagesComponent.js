import { ReactComponent as Ask } from './assets/ask.svg';
import { ReactComponent as Building } from './assets/building.svg';
import { ReactComponent as Blocked } from './assets/blocked.svg';
import { ReactComponent as Click } from './assets/click.svg';
import { ReactComponent as Forbidden } from './assets/forbidden.svg';
import { ReactComponent as Locked } from './assets/locked.svg';
import { ReactComponent as Nofound } from './assets/nofound.svg';
import { ReactComponent as Offline } from './assets/offline.svg';
import { ReactComponent as Sad } from './assets/sad.svg';
import { ReactComponent as Scripting } from './assets/scripting.svg';
import { ReactComponent as Timeout } from './assets/timeout.svg';
import { ReactComponent as Vehicle } from './assets/vehicle.svg';
import { ReactComponent as Unknow } from './assets/unknow.svg';

const listImagesAsComponent = {
  Ask,
  Blocked,
  Building,
  Click,
  Forbidden,
  Locked,
  Nofound,
  Offline,
  Sad,
  Scripting,
  Timeout,
  Vehicle,
  Unknow,
};

export default listImagesAsComponent;
