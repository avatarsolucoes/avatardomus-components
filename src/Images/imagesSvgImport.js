import ask from './assets/ask.svg';
import blocked from './assets/blocked.svg';
import building from './assets/building.svg';
import click from './assets/click.svg';
import forbidden from './assets/forbidden.svg';
import locked from './assets/locked.svg';
import nofound from './assets/nofound.svg';
import offline from './assets/offline.svg';
import sad from './assets/sad.svg';
import scripting from './assets/scripting.svg';
import timeout from './assets/timeout.svg';
import vehicle from './assets/vehicle.svg';
import unknow from './assets/unknow.svg';

const listImagesAsSvg = {
  ask,
  blocked,
  building,
  click,
  forbidden,
  locked,
  nofound,
  offline,
  sad,
  scripting,
  timeout,
  vehicle,
  unknow,
};

export default listImagesAsSvg;
