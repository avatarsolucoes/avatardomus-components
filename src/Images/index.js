import imgMessage from './message';
import imgIcon from './icon';
import imgForm from './form';

export { imgMessage, imgIcon, imgForm };

export { getKeyImage } from './utils';
