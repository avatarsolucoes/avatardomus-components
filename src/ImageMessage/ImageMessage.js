import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import { getImageByType } from '../Images';

import css from './image-message.css';

export default function ImageMessage(props) {
  const { type, text, className, children, src, size, raw } = props;

  const c = {
    main: cx(css.imageMessage, 'image-message', className),
    img: cx(css.imageContent, 'image-content'),
    text: cx(css.messageText, 'message-text'),
  };
  const element = children || (text ? <p className={c.text}>{text}</p> : null);

  const sizes = size ? { width: size, height: size } : {};

  function renderImage() {
    if (!src) {
      const Img = getImageByType(type, false);
      return <Img {...sizes} />;
    }
    const source = !raw ? src : getImageByType(type, true);
    return <img src={source} alt={type} title={type} className={c.img} styles={sizes} />;
  }

  return (
    <div className={c.main}>
      {renderImage()}
      {element}
    </div>
  );
}

ImageMessage.propTypes = {
  type: PropTypes.string,
  text: PropTypes.any,
  children: PropTypes.any,
  className: PropTypes.string,
  src: PropTypes.string,
  size: PropTypes.number,
  raw: PropTypes.bool,
};

ImageMessage.defaultProps = {
  type: 'unknow',
  children: null,
  text: '',
  className: null,
  src: null,
  size: null,
  raw: false,
};
