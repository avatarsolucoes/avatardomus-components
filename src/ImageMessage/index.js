import ImageMessage from './ImageMessage';

export { ImageMessage };
