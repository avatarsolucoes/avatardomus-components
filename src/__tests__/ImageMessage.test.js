import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow } from '../../setupEnzyme';

import { ImageMessage } from '../ImageMessage';
import { svgComponentList, getKeyImage } from '../Images';

const src = 'https://avatarsolucoesdigitais.com.br/images/logoh100.svg';

describe('ImageMessage', () => {
  it('is truthy ImageMessage', () => {
    expect(ImageMessage).toBeTruthy();
  });

  const namesList = getKeyImage(svgComponentList);
  it(`Testando namesList`, () => {
    const imagesMount = namesList.map((imgKey) => {
      return {
        withouText: renderer.create(<ImageMessage type={imgKey} className="teste1" />).toJSON(),
        withText: renderer.create(<ImageMessage text="test" type={imgKey} size={100} />).toJSON(),
        raw: renderer.create(<ImageMessage raw type={imgKey} />).toJSON(),
      };
    });
    expect(imagesMount).toMatchSnapshot();
  });

  it(`Testando with src`, () => {
    const wrapper = {
      raw: renderer.create(<ImageMessage raw src={src} />).toJSON(),
      src: renderer.create(<ImageMessage src={src} />).toJSON(),
    };
    expect(wrapper).toMatchSnapshot();
  });
});
