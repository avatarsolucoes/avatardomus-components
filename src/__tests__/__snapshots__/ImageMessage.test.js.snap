// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageMessage Test namesList 1`] = `
Array [
  Object {
    "raw": <div
      className="image-message"
    >
      <svg>
        ask.svg
      </svg>
    </div>,
    "withText": <div
      className="image-message"
    >
      <svg
        height={100}
        width={100}
      >
        ask.svg
      </svg>
      <p
        className="message-text"
      >
        test
      </p>
    </div>,
    "withouText": <div
      className="image-message teste1"
    >
      <svg>
        ask.svg
      </svg>
    </div>,
  },
  Object {
    "raw": <div
      className="image-message"
    >
      <svg>
        blocked.svg
      </svg>
    </div>,
    "withText": <div
      className="image-message"
    >
      <svg
        height={100}
        width={100}
      >
        blocked.svg
      </svg>
      <p
        className="message-text"
      >
        test
      </p>
    </div>,
    "withouText": <div
      className="image-message teste1"
    >
      <svg>
        blocked.svg
      </svg>
    </div>,
  },
  Object {
    "raw": <div
      className="image-message"
    >
      <svg>
        building.svg
      </svg>
    </div>,
    "withText": <div
      className="image-message"
    >
      <svg
        height={100}
        width={100}
      >
        building.svg
      </svg>
      <p
        className="message-text"
      >
        test
      </p>
    </div>,
    "withouText": <div
      className="image-message teste1"
    >
      <svg>
        building.svg
      </svg>
    </div>,
  },
  Object {
    "raw": <div
      className="image-message"
    >
      <svg>
        click.svg
      </svg>
    </div>,
    "withText": <div
      className="image-message"
    >
      <svg
        height={100}
        width={100}
      >
        click.svg
      </svg>
      <p
        className="message-text"
      >
        test
      </p>
    </div>,
    "withouText": <div
      className="image-message teste1"
    >
      <svg>
        click.svg
      </svg>
    </div>,
  },
  Object {
    "raw": <div
      className="image-message"
    >
      <svg>
        forbidden.svg
      </svg>
    </div>,
    "withText": <div
      className="image-message"
    >
      <svg
        height={100}
        width={100}
      >
        forbidden.svg
      </svg>
      <p
        className="message-text"
      >
        test
      </p>
    </div>,
    "withouText": <div
      className="image-message teste1"
    >
      <svg>
        forbidden.svg
      </svg>
    </div>,
  },
  Object {
    "raw": <div
      className="image-message"
    >
      <svg>
        locked.svg
      </svg>
    </div>,
    "withText": <div
      className="image-message"
    >
      <svg
        height={100}
        width={100}
      >
        locked.svg
      </svg>
      <p
        className="message-text"
      >
        test
      </p>
    </div>,
    "withouText": <div
      className="image-message teste1"
    >
      <svg>
        locked.svg
      </svg>
    </div>,
  },
  Object {
    "raw": <div
      className="image-message"
    >
      <svg>
        nofound.svg
      </svg>
    </div>,
    "withText": <div
      className="image-message"
    >
      <svg
        height={100}
        width={100}
      >
        nofound.svg
      </svg>
      <p
        className="message-text"
      >
        test
      </p>
    </div>,
    "withouText": <div
      className="image-message teste1"
    >
      <svg>
        nofound.svg
      </svg>
    </div>,
  },
  Object {
    "raw": <div
      className="image-message"
    >
      <svg>
        offline.svg
      </svg>
    </div>,
    "withText": <div
      className="image-message"
    >
      <svg
        height={100}
        width={100}
      >
        offline.svg
      </svg>
      <p
        className="message-text"
      >
        test
      </p>
    </div>,
    "withouText": <div
      className="image-message teste1"
    >
      <svg>
        offline.svg
      </svg>
    </div>,
  },
  Object {
    "raw": <div
      className="image-message"
    >
      <svg>
        sad.svg
      </svg>
    </div>,
    "withText": <div
      className="image-message"
    >
      <svg
        height={100}
        width={100}
      >
        sad.svg
      </svg>
      <p
        className="message-text"
      >
        test
      </p>
    </div>,
    "withouText": <div
      className="image-message teste1"
    >
      <svg>
        sad.svg
      </svg>
    </div>,
  },
  Object {
    "raw": <div
      className="image-message"
    >
      <svg>
        scripting.svg
      </svg>
    </div>,
    "withText": <div
      className="image-message"
    >
      <svg
        height={100}
        width={100}
      >
        scripting.svg
      </svg>
      <p
        className="message-text"
      >
        test
      </p>
    </div>,
    "withouText": <div
      className="image-message teste1"
    >
      <svg>
        scripting.svg
      </svg>
    </div>,
  },
  Object {
    "raw": <div
      className="image-message"
    >
      <svg>
        timeout.svg
      </svg>
    </div>,
    "withText": <div
      className="image-message"
    >
      <svg
        height={100}
        width={100}
      >
        timeout.svg
      </svg>
      <p
        className="message-text"
      >
        test
      </p>
    </div>,
    "withouText": <div
      className="image-message teste1"
    >
      <svg>
        timeout.svg
      </svg>
    </div>,
  },
  Object {
    "raw": <div
      className="image-message"
    >
      <svg>
        vehicle.svg
      </svg>
    </div>,
    "withText": <div
      className="image-message"
    >
      <svg
        height={100}
        width={100}
      >
        vehicle.svg
      </svg>
      <p
        className="message-text"
      >
        test
      </p>
    </div>,
    "withouText": <div
      className="image-message teste1"
    >
      <svg>
        vehicle.svg
      </svg>
    </div>,
  },
  Object {
    "raw": <div
      className="image-message"
    >
      <svg>
        unknow.svg
      </svg>
    </div>,
    "withText": <div
      className="image-message"
    >
      <svg
        height={100}
        width={100}
      >
        unknow.svg
      </svg>
      <p
        className="message-text"
      >
        test
      </p>
    </div>,
    "withouText": <div
      className="image-message teste1"
    >
      <svg>
        unknow.svg
      </svg>
    </div>,
  },
]
`;

exports[`ImageMessage Test with src 1`] = `
Object {
  "raw": <div
    className="image-message"
  >
    <img
      alt="unknow"
      className="image-content"
      src={
        Object {
          "ReactComponent": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "__esModule": true,
          "default": "unknow.svg",
        }
      }
      styles={Object {}}
      title="unknow"
    />
  </div>,
  "src": <div
    className="image-message"
  >
    <img
      alt="unknow"
      className="image-content"
      src="https://avatarsolucoesdigitais.com.br/images/logoh100.svg"
      styles={Object {}}
      title="unknow"
    />
  </div>,
}
`;
